### PR TITLE
chore(all): nx

### PR DESCRIPTION
**Pull request by julianpoy**
Because:

* We want to introduce nx to the repository

This commit:

* Adds nx with working builds

Closes: [FXA-7341](https://mozilla-hub.atlassian.net/browse/FXA-7341)

[FXA-7341]: https://mozilla-hub.atlassian.net/browse/FXA-7341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jest.preset.js
Comment (pdehaan) [inline jest.preset.js:4]: Nit: indentation, semi-colons, and EOL are a bit off in this file.

File: .eslintrc.json
Comment (pdehaan) [inline .eslintrc.json:13]: Q: Do we still want to extend `"plugin:fxa/recommended"` config and/or add the `ignorePatterns[]`?
Or are these all handled in the individual .eslintrc config files now?

File: .eslintrc.json
Comment (pdehaan) [inline .eslintrc.json:11]: nit: Maybe a-z these to keep the `@typescript-eslint/*` rules clustered. 🤷

File: jest.preset.js
Comment (pdehaan) [reply jest.preset.js:4]: Also, should this be a .ts file to match jest.config.ts above?

File: nx.json
Comment (pdehaan) [inline nx.json:40]: Interesting! What do these do?
/runs to check Nx docs.

File: packages/fxa-auth-client/package.json
Comment (pdehaan) [inline packages/fxa-auth-client/package.json:16]: Q: Why remove the `lint` script from each package?

I see we added it above in nx.json
https://github.com/mozilla/fxa/blob/a23877ae4c03b62f0a3426b4f7cc62b93a5a09ab/nx.json#L17-L19

And seemingly removed from 6 packages/*/package.json files in this PR, but maybe not _all_ package.json files?

```sh
git grep -n '"lint": ' packages/**/package.json | cat

packages/123done/package.json:41:    "lint": "eslint .",
packages/browserid-verifier/package.json:46:    "lint": "eslint .",
packages/fortress/package.json:39:    "lint": "npm-run-all --parallel lint-eslint",
packages/fxa-admin-panel/package.json:13:    "lint": "eslint .",
packages/fxa-admin-server/package.json:11:    "lint": "eslint .",
packages/fxa-auth-server/package.json:19:    "lint": "eslint . .storybook --ext .js,.ts",
packages/fxa-content-server/package.json:13:    "lint": "eslint app server tests --cache",
packages/fxa-customs-server/package.json:21:    "lint": "eslint .",
packages/fxa-geodb/package.json:12:    "lint": "eslint .",
packages/fxa-profile-server/package.json:7:    "lint": "eslint .",
packages/fxa-react/package.json:21:    "lint": "eslint . .storybook",
packages/fxa-shared/package.json:75:    "lint": "eslint .",
```

File: jest.preset.js
Comment (julianpoy) [reply jest.preset.js:4]: Changed PR to draft.

File: .eslintrc.json
Comment (julianpoy) [reply .eslintrc.json:13]: We want to remove `"plugin:fxa/recommended"` from our ecosystem entirely.

I should probably throw some ignorePatterns in here.

File: packages/fxa-auth-client/package.json
Comment (julianpoy) [reply packages/fxa-auth-client/package.json]: This should all be resolved now.

**Comment by julianpoy** [comment]
There are a few files conflicting. Given that this touches so many files, I'll hold off on rebasing until we're ready to merge.

File: nx.json
Comment (bbangert) [reply nx.json:40]: These are for the integrated repository style, which works with tsconfig such that we can organize libraries/apps in directories that all share the root node_modules rather than individual packages like the legacy /packages style.

**Review by bbangert — APPROVED** [review]
LGTM! I presume we'll need to test/watch CI for awhile after this to make sure its covering what we believe its covering?

File: nx.json
Comment (dschom) [inline nx.json:29]: Perhaps we don't want to exclude tests? Scenarios where this may matter, a test could have pick of up typescript error, or a test could have logic changing which we wouldn't want to cache.

File: nx.json
Comment (dschom) [inline nx.json:8]: Add test-unit and test-integration? These are targets used by CI and likely by developers too.

**Review by dschom — APPROVED** [review]

File: nx.json
Comment (julianpoy) [reply nx.json]: Done

File: nx.json
Comment (julianpoy) [reply nx.json:29]: This is the production cache, which is an additional definition on top of the default cache for production-only targets (currently only build).

**Pull request merged**
